### DISCUSSION TITLE
logrotate

### DISCUSSION THREAD

**Discussion by str** [opcodesio/log-viewer#138] [Ideas]
Is it possible to read logrotate files?

**Comment by arukompas** [comment]
Currently only Laravel logs are supported. If they're being rotated, they should still be readable - just point to them with the `'include_paths'` config option.

**Reply by str** [reply]
You are right, that I can load  logrotated files if I switch `'include_files' => ['*.log']` to `'include_files' => ['*']`.  In this case I could see the logrotated files

- laravel.log
- laravel.log.1
- laravel.log.2

 I've seen, more ofthen than not, logrotate to gzip the files.  So the files would be 

- laravel.log
- laravel.log.1.gz
- laravel.log.2.gz

It would be great to read gz files by default.

**Reply by arukompas** [reply]
That's indeed missing at the moment! :) I wonder how efficient it is to read a gzipped log file on every request... 🤔

**Reply by str** [reply]
Gzip has a very low CPU usage. It is very common to use it to [compress every single HTTP request.](https://en.wikipedia.org/wiki/HTTP_compression).  Of course a benchmark could get hard data to confirm.

Gzipped logs are archived, and will not be updated anymore (until rotated). I think caching the unziped content so the logviewer would read the cache if already unzipped could solve any inefficiency
